mountpoint-s3/src/fs.rs

```diff
@@ -370,6 +370,33 @@ where
         })
     }
 
+    pub async fn mkdir(
```

Comment:
This is *very* similar to `mknod`, but I assume we want to keep 1-1 mapping between methods in `S3Filesystem` and `S3FuseFilesystem`.